test/Microsoft.AspNet.FileProviders.Combined.Tests/CombinedFileProviderTests.cs

```diff
@@ -0,0 +1,250 @@
+                    folderBFile1,
+                    folderCFile3
+                    )
+                );
+
+            // Act
+            var files = provider.GetDirectoryContents("FolderC/");
+
+            // Assert
+            Assert.NotNull(files);
+            Assert.True(files.Exists);
+            Assert.Collection(files.OrderBy(f => f.Name, StringComparer.Ordinal),
+                file => Assert.Equal(folderCFile3, file));
+        }
+
+        [Fact]
+        public void Watch_ReturnsNoopChangeToken_IfNoFileProviderSpecified()
+        {
+            // Arrange
+            var provider = new CombinedFileProvider();
+
+            // Act
+            var changeToken = provider.Watch("DoesNotExist*Pattern");
+
+            // Assert
+            Assert.NotNull(changeToken);
+            Assert.False(changeToken.ActiveChangeCallbacks);
+        }
+
+        [Fact]
+        public void Watch_ReturnsNoopChangeToken_IfNoWatcherReturnedByFileProviders()
+        {
+            // Arrange
+            var provider = new CombinedFileProvider(
+                new MockFileProvider()
+                );
+
+            // Act
+            var changeToken = provider.Watch("DoesntExist*Pattern");
+
+            // Assert
+            Assert.NotNull(changeToken);
+            Assert.False(changeToken.ActiveChangeCallbacks);
+        }
+
+        [Fact]
+        public void Watch_CombinedChangeToken()
+        {
+            // Arrange
+            var firstChangeToken = new MockChangeToken();
+            var secondChangeToken = new MockChangeToken();
+            var thirdChangeToken = new MockChangeToken();
+            var provider = new CombinedFileProvider(
+                new MockFileProvider(
+                    new KeyValuePair<string, IChangeToken>("pattern", firstChangeToken),
+                    new KeyValuePair<string, IChangeToken>("2ndpattern", secondChangeToken)),
+                new MockFileProvider(new KeyValuePair<string, IChangeToken>("pattern", thirdChangeToken)));
+
+            // Act
+            var changeToken = provider.Watch("pattern");
+
+            // Assert
+            Assert.NotNull(changeToken);
+            Assert.True(changeToken.ActiveChangeCallbacks);
+            Assert.False(changeToken.HasChanged);
+
+            // HasChanged update
+            // first change token
+            firstChangeToken.HasChanged = true;
+            Assert.True(changeToken.HasChanged);
+            firstChangeToken.HasChanged = false;
+            // second change token
+            secondChangeToken.HasChanged = true;
+            Assert.False(changeToken.HasChanged);
+            secondChangeToken.HasChanged = false;
+            // third change token
+            thirdChangeToken.HasChanged = true;
+            Assert.True(changeToken.HasChanged);
+            thirdChangeToken.HasChanged = false;
+
+            // Register callback
+            Assert.Equal(0, firstChangeToken.Callsback.Count);
```

Comment:
Could you split these into two tests? One's dealing with `HasChanged` and the other with callbacks.